main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -159,30 +181,31 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
     * Performs name resolution on the given enum `e0` in the given namespace `ns0`.
     */
   def resolve(e0: NamedAst.Enum, ns0: Name.NName, root: NamedAst.Root)(implicit flix: Flix): Validation[ResolvedAst.Enum, ResolutionError] = {
-    val tparamsVal = traverse(e0.tparams)(p => Params.resolve(p, ns0, root))
-    val casesVal = traverse(e0.cases) {
-      case (name, NamedAst.Case(enum, tag, tpe)) =>
+    traverse(e0.tparams)(p => Params.resolve(p, ns0, root)).flatMap {
+      tparams =>
+        val tconstrs = tparams.flatMap(tparam => tparam.classes.map(clazz => TypedAst.TypeConstraint(clazz, tparam.tpe)))
```

Comment:
This seems to suggest to me that enums can enforced that their type variables belong to certain type classes.

I have not seen that before. Can that be done in other languages and is it useful? (I could imagine it being useful).

Reply:
Whether it's possible in other languages, I'm not sure.

Ultimately, it's equivalent to putting the type constraint everywhere the enum is used:

```flix
enum E[a] {
    case C(a)
}
def f[a : Eq](x: E[a], y: E[a]): Bool = ???
def g[a: Eq](x: E[a]): E[a] = ???
```
vs
```flix
enum E[a : Eq] {
    case C(a)
}
def f(x: E[a], y: E[a]): Bool = ???
def g(x: E[a]): E[a] = ???
```

(Need to check that this actually works)